feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/transaction/history/TransactionHistoryAdapter.kt

```diff
@@ -70,32 +76,49 @@ class TransactionHolder(
 
     fun bind(item: OperationModel, handler: TransactionHistoryAdapter.Handler) {
         with(containerView) {
-            header.text = item.header
+            bindHeader(item)
+
+            bindAmount(item)
 
-            valuePrimary.setTextColorRes(item.amountColorRes)
-            valuePrimary.text = item.amount
+            bindAmountDetails(item)
+            bindSubHeader(item)
 
-            bindFiatWithTime(item)
-            subHeader.text = item.subHeader
-            subHeader.ellipsize = item.subHeaderEllipsize
             icon.setIcon(item.operationIcon, imageLoader)
 
-            if (item.statusAppearance != OperationStatusAppearance.COMPLETED) {
-                status.makeVisible()
-                status.setImageResource(item.statusAppearance.icon)
-                status.setImageTintRes(item.statusAppearance.statusIconTint)
-            } else {
-                status.makeGone()
-            }
+            bindStatus(item)
 
-            setOnClickListener { handler.transactionClicked(item) }
+            setOnClickListener { handler.transactionClicked(item.id) }
```

Comment:
This is to avoid potentially outdated model arriving to `handler.transactionClicked` since we do not re-set onClickListener. We only use `item.id` anyway